main.tf

```diff
@@ -4,10 +4,54 @@ data "aws_eks_cluster" "this" {
 
 data "aws_region" "current" {}
 
+locals {
+  argocd_enabled     = length(var.argocd) > 0 ? 1 : 0
+  grafana_enabled    = var.grafana_enabled ? 1 : 0
+  prometheus_enabled = var.prometheus_enabled ? 1 : 0
+  thanos_enabled     = var.thanos_enabled ? 1 : 0
+  storage            = var.thanos_storage == "s3" ? 0 : 1
```

Comment:
is that mean that source-specific resources will be created independently of Thanos installation? I don't see any logic between thanos_enabled and thanos_storage?  looks like s3 bucket will be created always by default

Reply:
it's for choose minio or s3 backend and resource "kubernetes_secret" "s3_objstore"  or resource "kubernetes_secret" "thanos_objstore"  will be created